crates/hotshot-stake-table/src/mt_based.rs

```diff
@@ -86,7 +88,18 @@ impl<K: Key> StakeTableScheme for StakeTable<K> {
         self.mapping.contains_key(key)
     }
 
-    fn lookup(
+    fn lookup(&self, version: SnapshotVersion, key: &K) -> Result<Self::Amount, StakeTableError> {
+        let root = Self::get_root(self, version)?;
+        match self.mapping.get(key) {
+            Some(index) => {
+                let branches = to_merkle_path(*index, self.height);
+                root.simple_lookup(self.height, &branches)
+            }
+            None => Err(StakeTableError::KeyNotFound),
+        }
+    }
+
+    fn lookup_with_proof(
```

Comment:
We can add a private func `internal_lookup(self, version, key, is_simple_lookup: bool)` to remove code redundancy.

Reply:
It already calls `lookup_pos` internally, and the rest is just fetching items.

Reply:
Idk why I can't reply to the comment about compression. But currently the key is over G2, thus the compressed form is already 2 Fq elements + 1 bit. Which requires 3 Fr elements to hold. And I believe the canonical serialization is already doing the job.

Reply:
cc @alxiong